CONTRIBUTING.md

```diff
@@ -1,197 +1,221 @@
-6. Verify that CI passes all [status checks](https://help.github.com/articles/about-status-checks/). Fix if needed.
+6. Verify that CI passes all [status checks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks).
+   Fix if needed.
 7. Wait for other developers to review your code and update code as needed.
 8. Once reviewed and approved, a RAPIDS developer will merge your pull request.
 
-Remember, if you are unsure about anything, don't hesitate to comment on issues
-and ask for clarifications!
+If you are unsure about anything, don't hesitate to comment on issues and ask for clarification!
 
 ### Seasoned developers
 
-Once you have gotten your feet wet and are more comfortable with the code, you
-can look at the prioritized issues for our next release in our [project boards](https://github.com/rapidsai/cudf/projects).
+Once you have gotten your feet wet and are more comfortable with the code, you can look at the
+prioritized issues for our next release in our
+[project boards](https://github.com/rapidsai/cudf/projects).
 
-> **Pro Tip:** Always look at the release board with the highest number for
-issues to work on. This is where RAPIDS developers also focus their efforts.
+**Note:** Always look at the release board that is
+[currently under development](https://docs.rapids.ai/maintainers) for issues to work on. This is
+where RAPIDS developers also focus their efforts.
 
-Look at the unassigned issues, and find an issue to which you are comfortable
-contributing. Start with _Step 3_ above, commenting on the issue to let
-others know you are working on it. If you have any questions related to the
-implementation of the issue, ask them in the issue instead of the PR.
-
-## Setting Up Your Build Environment
-
-The following instructions are for developers and contributors to cuDF OSS development. These instructions are tested on Linux Ubuntu 16.04 & 18.04. Use these instructions to build cuDF from source and contribute to its development.  Other operating systems may be compatible, but are not currently tested.
+Look at the unassigned issues, and find an issue to which you are comfortable contributing. Start
+with _Step 3_ above, commenting on the issue to let others know you are working on it. If you have
+any questions related to the implementation of the issue, ask them in the issue instead of the PR.
 
+## Setting up your build environment
 
+The following instructions are for developers and contributors to cuDF development. These
+instructions are tested on Ubuntu Linux LTS releases. Use these instructions to build cuDF from
+source and contribute to its development. Other operating systems may be compatible, but are not
+currently tested.
 
 ### General requirements
 
 Compilers:
 
-* `gcc`     version 9.3+
-* `nvcc`    version 11.5+
-* `cmake`   version 3.20.1+
+* `gcc` version 9.3+
+* `nvcc` version 11.5+
+* `cmake` version 3.20.1+
```

Comment:
This is mainly an academic curiosity, but info like this goes stale very quickly. I wonder if there is a way to programmatically query it from a cmake file? @robertmaynard any ideas?

Reply:
CMake gets picked up from the conda env, FWIW..

Reply:
I would prefer to point to an external source of this information (maybe the conda environment) and not specify versions here at all. (Unfortunately, our conda env doesn't necessarily specify our _minimal_ requirements - just a working configuration that is highly pinned.) I thought about this but I don't have any clever ideas for automating it.

Reply:
The root level CMakeLists.txt has a `cmake_minimum_required(VERSION 3.20.1 FATAL_ERROR)`. I expect we could run a script in our CI script process the makes sure the contributing.md and this line stay in sync.

Reply:
@jrhemstad Is an action item needed here to unblock/approve this PR or pursue in a follow-up PR? I'm inclined to punt on this as nice to have but out of scope.

Reply:
It was just a "nice to have" curiosity. Definitely not a blocker.